pkg/frontend/transport/handler.go

```diff
@@ -135,17 +134,24 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}()
 
 	// Buffer the body for later use to track slow queries.
-	var buf bytes.Buffer
-	r.Body = http.MaxBytesReader(w, r.Body, f.cfg.MaxBodySize)
-	r.Body = io.NopCloser(io.TeeReader(r.Body, &buf))
+	var err error
+	r.Body, err = newReadCloseSeeker(http.MaxBytesReader(w, r.Body, f.cfg.MaxBodySize))
```

Comment:
Why do we need to call `newReadCloseSeeker()` here, given few lines below we also call `withSeekerBody()`?

Reply:
I just prefer to be explicit for the code reader. I want any reader of this method explicitly see that we're using a seeker for the body. This would make the `withSeekerBody` not go into the replacing path (which is why I didn't detect the issue there)

Reply:
Ok, that's fine.